### DISCUSSION TITLE
How to create/deploy custom workflows (Kubernetes)

### DISCUSSION THREAD

**Discussion by syntastical** [StackStorm/st2#6211] [Documentation]
Hey all,

I recently started playing around with StackStorm deployed from the official Helm chart in Kubernetes.  I'm getting a bit confused about the process for defining and using new custom workflows.  I created a new workflow in the web ui, and after saving, it can run the workflow.  However, the workflow execution fails because the runner cannot find the workflow file, with error:
```json
{
  "error": "[Errno 2] No such file or directory: '/opt/stackstorm/packs/default/actions/workflows/test.yaml'",
  "traceback": "  File \"/opt/stackstorm/st2/lib/python3.8/site-packages/st2actions/container/base.py\", line 132, in _do_run\n    (status, result, context) = runner.run(action_params)\n  File \"/opt/stackstorm/st2/lib/python3.8/site-packages/orquesta_runner/orquesta_runner.py\", line 125, in run\n    return self.start_workflow(action_parameters)\n  File \"/opt/stackstorm/st2/lib/python3.8/site-packages/orquesta_runner/orquesta_runner.py\", line 129, in start_workflow\n    wf_def = self.get_workflow_definition(self.entry_point)\n  File \"/opt/stackstorm/st2/lib/python3.8/site-packages/orquesta_runner/orquesta_runner.py\", line 49, in get_workflow_definition\n    with open(entry_point, \"r\") as def_file:\n"
}
```

After doing some research, it seems like I either need to 
1. share a volume across all my runners and the api so that the change is available after save.
2. create the workflow in the editor, copy the yaml, add it to my helm chart, then deploy.

If case #1 is true, will StackStorm automatically pick up the change after save, or do I need to do anything else after saving?
If case #2 is true, it would be nice to
1.  document this flow (if the documentation exists, sorry I really tried to look).  
2. potentially disable the run button if there isn't a shared volume and the workflow was created in the ui or not present on the runners.
3. have the api push the config to the runners or store in a database and have the runners pull it.
4. have some ui hint that tells users the configuration might on exist on the runner unless a shared volume is used.

Thanks for any help

**Comment by fdrab** [comment]
1, ST2 should auto-pick up new flows created in the UI. The UI is far from perfect and at times it can brick your yaml definitions to a point that you can't reload the flow in the UI without manually editing the file to fix the yaml.

Any manual modifications to flows need to be applied by reloading the flow manually using packs.load action (as far as I know)

**Reply by syntastical** [reply]
Odd, the yaml file gets created on my API instance just fine but when I run the workflow it fails with the error above.  When I check my runners, they don't have the workflow yaml file present.  Any idea what might be wrong with my configuration?

**Reply by chain312** [reply]
This method is recommended to store files.
![image](https://github.com/StackStorm/st2/assets/30886624/0b8fb3f5-166e-457e-98e4-3d3348b92e8c)
